velox/functions/sparksql/tests/CompareTests.cpp

```diff
@@ -0,0 +1,298 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/common/base/tests/GTestUtils.h"
+#include "velox/functions/sparksql/tests/SparkFunctionBaseTest.h"
+
+#include <velox/vector/SimpleVector.h>
+
+namespace facebook::velox::functions::sparksql::test {
+namespace {
+static constexpr double kInf = std::numeric_limits<double>::infinity();
+static constexpr float kInfF = std::numeric_limits<float>::infinity();
+static constexpr auto kNaN = std::numeric_limits<double>::quiet_NaN();
+class CompareTest : public SparkFunctionBaseTest {
+ protected:
+  template <typename T>
+  std::optional<bool> equaltonullsafe(std::optional<T> a, std::optional<T> b) {
+    return evaluateOnce<bool>("equalnullsafe(c0, c1)", a, b);
```

Comment:
Duplicated with `CompareNullSafeTests.cpp`?

Reply:
`CompareNullSafeTests.cpp` should be removed.